documentation-website/Writerside/snippets/exposed-dao/src/main/kotlin/org/example/App.kt

```diff
@@ -1,13 +1,138 @@
 package org.example
 
+import org.example.entities.*
+import org.example.tables.*
+import org.jetbrains.exposed.dao.id.CompositeID
 import org.jetbrains.exposed.sql.*
+import org.jetbrains.exposed.sql.SqlExpressionBuilder.eq
 import org.jetbrains.exposed.sql.transactions.transaction
+import java.util.*
 
 fun main() {
     Database.connect("jdbc:h2:mem:test", driver = "org.h2.Driver")
 
     transaction {
         addLogger(StdOutSqlLogger)
-        SchemaUtils.create(StarWarsFilms)
+        createFilms()
+        createDirectors()
+        createUsersAndRatings()
     }
 }
+
+fun createFilms() {
```

Comment:
Would it be feasible to separate these functions into separate files? Or is the plan that code in these functions should never be added to? Because if a new line of code is added, it potentially breaks all code blocks that reference specific sections by line.

At least that happened when I tried to add a line for the fixed findById(CompositeID) code & run the project. Any code after from `createUsersAndRatings()` that was used in a block was displaced by a line and either threw an error or showed wrong code.

Reply:
Sure! I found that we could also use the `include-symbol` attribute on the `code-block`s with functions or variables, so I separated the examples within the new `examples` folder and assigned variables to almost everything. Let me know if you think this is a good approach.

Reply:
Yup, that seems like a better alternative and we can try to use symbols whenever possible.

If we do have to use lines somewhere, because functions or variables are not possible, maybe we can try to have a system where we leave a comment on the line above, noting which topic uses that line. That way anybody who adds/deletes a line in a function can more easily see where to check the code blocks? I could also see people forgetting to do this, so at least this approach minimizes the need to use lines directly.

Reply:
☑️ Added some instructions in the README file in `snippets` on how to reference code from examples and commented on the files where we use direct line references.